project/src/main/kotlin/org/ionproject/core/calendar/CalendarData.kt

```diff
@@ -0,0 +1,286 @@
+package org.ionproject.core.calendar
+
+import org.ionproject.core.calendar.category.CategoryRepo
+import org.ionproject.core.calendar.icalendar.CalendarComponent
+import org.ionproject.core.calendar.icalendar.Event
+import org.ionproject.core.calendar.icalendar.Journal
+import org.ionproject.core.calendar.icalendar.Todo
+import org.ionproject.core.calendar.icalendar.properties.components.change_management.DateTimeCreated
+import org.ionproject.core.calendar.icalendar.properties.components.change_management.DateTimeStamp
+import org.ionproject.core.calendar.icalendar.properties.components.datetime.DateTimeDue
+import org.ionproject.core.calendar.icalendar.properties.components.datetime.DateTimeEnd
+import org.ionproject.core.calendar.icalendar.properties.components.datetime.DateTimeStart
+import org.ionproject.core.calendar.icalendar.properties.components.descriptive.Attachment
+import org.ionproject.core.calendar.icalendar.properties.components.descriptive.Categories
+import org.ionproject.core.calendar.icalendar.properties.components.descriptive.Description
+import org.ionproject.core.calendar.icalendar.properties.components.descriptive.Summary
+import org.ionproject.core.calendar.icalendar.properties.components.recurrence.RecurrenceRule
+import org.ionproject.core.calendar.icalendar.properties.components.relationship.UniqueIdentifier
+import org.ionproject.core.calendar.icalendar.types.*
+import org.ionproject.core.calendar.language.LanguageRepo
+import org.jdbi.v3.core.mapper.RowMapper
+import org.jdbi.v3.core.statement.StatementContext
+import org.springframework.stereotype.Component
+import java.sql.ResultSet
+import java.time.OffsetDateTime
+import org.ionproject.core.calendar.icalendar.types.Date as DateType
+
+object CalendarData {
+    private const val CALENDAR_COMPONENT_ABBREVIATION = "cmp"
+    private const val CALENDAR_COMPONENT = "dbo.v_ComponentsAll $CALENDAR_COMPONENT_ABBREVIATION"
+
+    private const val UID_COLUMN = "uid"
```

Comment:
It's very hard for someone who hasn't been working on this class to read it and even harder to debug it because of all those references a person has to navigate to understand the whole code.

Reply:
How is it hard? It's the same as an hardcoded string. The variable names are highly suggestive and don't really require that you check the content most of the time.

Reply:
You are right about that. Its my own personal opinion and my compulsive need to check whats stored in the reference, therefore we can ignore this for now.